google/cloud/pubsub/samples/topic_admin_samples.cc

```diff
@@ -28,6 +30,36 @@ using TopicAdminCommand =
     std::function<void(::google::cloud::pubsub_admin::TopicAdminClient,
                        std::vector<std::string> const&)>;
 
+// Delete all topics created with that include "cloud-cpp-samples". Ignore any
+// failures. If multiple tests are cleaning up topics in parallel, then the
+// delete call might fail.
+void CleanupTopics(
+    google::cloud::pubsub_admin::TopicAdminClient& topic_admin_client,
+    std::string const& project_id, absl::Time time_now) {
+  std::cout << "Cleaning up old topics...\n";
+  auto const parent = google::cloud::Project(project_id).FullName();
+  for (auto& topic : topic_admin_client.ListTopics(parent)) {
+    if (!topic) continue;
+    auto topic_name = topic->name();
+    std::string keyword = "cloud-cpp-samples";
+    if (!absl::StrContains(topic->name(), keyword)) continue;
+    // Extract the date from the resource name which is in the format
+    // `*-cloud-cpp-samples-YYYY-MM-DD-`.
+    auto date =
+        topic_name.substr(topic_name.find(keyword) + keyword.size() + 1, 10);
+
+    absl::TimeZone nyc;
+    if (!absl::LoadTimeZone("America/New_York", &nyc)) continue;
+    absl::CivilDay day;
+    if (absl::ParseCivilTime(date, &day) &&
+        absl::FromCivil(day, nyc) < time_now - absl::Hours(36)) {
```

Comment:
Consider using `internal::FormatUtcDate(t)`, just like `RandomTopicId()` does.  Mixing timezones seems bad.

(See `google/cloud/spanner/testing/cleanup_stale_databases.cc` for something similar.)

Reply:
Oops, didn't realize. Changed to using UTC timezone